defi/src/getSimpleChainDataset.ts

```diff
@@ -1,27 +1,20 @@
 import { wrap, IResponse, errorResponse } from "./utils/shared";
 import { getChainDisplayName, chainCoingeckoIds, transformNewChainName } from "./utils/normalizeChain";
-import { getHistoricalTvlForAllProtocols } from "./storeGetCharts";
+import { getCachedHistoricalTvlForAllProtocols, getHistoricalTvlForAllProtocols } from "./storeGetCharts";
 import { formatTimestampAsDate, getClosestDayStartTimestamp, secondsInHour } from "./utils/date";
 import { buildRedirectR2, getR2, storeDatasetR2 } from "./utils/r2";
 
-const handler = async (event: AWSLambda.APIGatewayEvent): Promise<IResponse> => {
-  const rawChain = decodeURI(event.pathParameters!.chain!);
-  const globalChain = rawChain === "All" ? null : getChainDisplayName(rawChain.toLowerCase(), true);
-  const params = event.queryStringParameters ?? {};
+export async function getSimpleChainDatasetInternal(rawChain: string, params: any = {}) {
   const categorySelected = params.category===undefined?undefined:decodeURI(params.category);
+  const globalChain = rawChain === "All" ? null : getChainDisplayName(rawChain.toLowerCase(), true);
 
   const sumDailyTvls = {} as {
     [ts: number]: {
       [protocol: string]: number;
     };
   };
 
-  let historicalProtocolTvlsData:Awaited<ReturnType<typeof getHistoricalTvlForAllProtocols>>
-  if(categorySelected === "Bridge"){
-    historicalProtocolTvlsData = await getHistoricalTvlForAllProtocols(true, false);
-  } else {
-    historicalProtocolTvlsData = JSON.parse((await getR2(`cache/getHistoricalTvlForAllProtocols/false-${categorySelected === undefined}.json`)).body!)
-  }
+  let historicalProtocolTvlsData:Awaited<ReturnType<typeof getHistoricalTvlForAllProtocols>>  = await getCachedHistoricalTvlForAllProtocols(categorySelected === "Bridge", false);
```

Comment:
isnt this missing the `categorySelected === undefined` bit?

Reply:
good catch, fixing it